### PR TITLE
Added new folder docs with config-options description on options.

### PR DESCRIPTION
**Pull request by ehrktia**
[issue-#367](https://github.com/hound-search/hound/issues/367)

Add documentation for all the config options

### CODE REVIEW

**Comment by salemhilal** [comment]
This is a good start! A few comments: 

1. Can you add an introduction that explains that these are the keys and values of the config file? It would also help to explain how the sections relate to each other (like how the git options relate to the config options).
2. We'll probably want a link in the README to this doc.
3. It might make sense to have a column for Defaults
4. The names for each option don't always match what they need to be in the actual config.json. For example, "MaxConcurrentIndexers" should be `max-concurrent-indexers`.
5. Can the title of each section use regular casing? For example, "Config Options" instead of "ConfigOption" and "Git Options" instead of "gitOptions"?

**Comment by ehrktia** [comment]
@salemhilal thanks for your time doing peer review , i have updated the readme and config with requested updates.Please check and let me know.

File: README.md [hunk 1]
```diff
@@ -38,6 +39,8 @@ the config found in [default-config.json](default-config.json):
 }
 ```
 
+
+complete list of available config options can be found under [config-options](docs/config-options.md).
```
Comment (salemhilal) [inline README.md:43]: Nit:
```suggestion
A complete list of available config options can be found [here](docs/config-options.md).  
```

File: docs/config-options.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+- [ConfigOptions](#configoptions)
+  * [Git options](#git-options)
+  * [SVN options](#svn-options)
+  * [URL options](#url-options)
+
+
+
+### ConfigOptions
+Various options available in `config.json` are used by hound.Comprehensive list of available options are provided below with some sane defaults.
```
Comment (salemhilal) [inline docs/config-options.md:9]: Nit:
```suggestion
Most of Hound's behavior is defined by a single `config.json` configuration file. A list of its available options are provided below, along with their defaults.
```

File: docs/config-options.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+- [ConfigOptions](#configoptions)
+  * [Git options](#git-options)
+  * [SVN options](#svn-options)
+  * [URL options](#url-options)
+
+
+
+### ConfigOptions
```
Comment (salemhilal) [inline docs/config-options.md:8]: Nit: 
```suggestion
# Config Options
```

File: docs/config-options.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+:------ | :----- | :-----
+max-concurrent-indexers | defines the total number of indexers required to be used for indexing code | 2
+health-check-uri |  health check url for hound | `/healthz`
+dbpath | absolute file path where the `config.json` file exists| `data`
+title | Title used for the application | Hound
+url-pattern | composed of base url and anchor values in form of key value pairs | n/a
+vcs-config | holds the version control config, default VCS used in Hound is git.Other options for VCS are svn,mercurial,bitbucket,hg, etc.Refer to `config-example.json` to get the list of vcs and usage. Below tables provide detailed options list of each type of vcs | git
+repos | holds the list of repos which are required to be indexed by Hound . Each Repo is added with reponame as a Json Key with options associated with repo as values similar to example provided in `config-example.json` | n/a
+
+#### Git options
```
Comment (salemhilal) [inline docs/config-options.md:23]:
```suggestion
## Git Options
```

File: docs/config-options.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+List of options associated with git vcs in repos
+
+GitOptions  | Description | Default Values
+:------ | :----- | :-----
+ms-between-polls | time interval to poll the repo url | 30s
+detect-ref    | used to determine branch |  master branch 
+ref | used to provide reference for the branch for repo| n/a
+
+#### SVN options
```
Comment (salemhilal) [inline docs/config-options.md:32]:
```suggestion
## SVN Options
```

File: docs/config-options.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+List of options available for SVN vcs in repos
+
+SvnOptions  | Descriptions| Default Values
+:------ | :-----| :-----
+username  | user name for the svn repo | n/a
+password | password to authenticate use for svn repo | n/a
+
+
+#### URL options
```
Comment (salemhilal) [inline docs/config-options.md:42]:
```suggestion
## URL Options 
```

**Review by salemhilal — COMMENTED** [review]
Sorry for the delay on this! I left a handful of minor fixes, mostly style-related. If you merge them in, I'm happy to merge this PR.

**Comment by salemhilal** [comment]
Actually, it looks like I can apply changes myself. Let me know if you're ok with them and I'll merge this branch in.

**Comment by ehrktia** [comment]
@salemhilal thank you very much am okay for this to be merged.

**Pull request merged**